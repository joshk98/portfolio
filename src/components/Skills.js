import React from "react";
import "../styles/skills.css";

const skillsData = [
  "javascript",
  "react",
  "html5",
  "css3",
  "figma",
  "nodejs",
  "git",
  "postgresql",
  "docker",
  "amazonwebservices",
  "wordpress",
];

const mappedSkillsData = skillsData.map((skill, index) => ({
  id: index + 1,
  src: `https://cdn.jsdelivr.net/gh/devicons/devicon/icons/${skill}/${skill}-original.svg`,
  alt: skill,
}));

function Skills() {
  return (
    <div className="skills-container">
      <div className="skills-container-images">
        {mappedSkillsData.map((skill) => (
          <div key={skill.id}>
            <img src={skill.src} alt={skill.alt} />
          </div>
        ))}
        <img
          src="https://cdn.worldvectorlogo.com/logos/asana-logo.svg"
          alt="asana"
        />
      </div>
    </div>
  );
}

export default Skills;
