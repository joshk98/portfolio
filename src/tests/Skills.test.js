import React from "react";
import { render } from "@testing-library/react";
import Skills from "../components/Skills";

describe("Skills", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<Skills />);

    expect(asFragment()).toMatchSnapshot();
  });
});
