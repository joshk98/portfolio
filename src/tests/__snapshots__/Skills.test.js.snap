// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills renders correctly 1`] = `
<DocumentFragment>
  <div
    class="skills-container"
  >
    <div
      class="skills-container-images"
    >
      <div>
        <img
          alt="javascript"
          src="https://cdn.jsdelivr.net/gh/devicons/devicon/icons/javascript/javascript-original.svg"
        />
      </div>
      <div>
        <img
          alt="react"
          src="https://cdn.jsdelivr.net/gh/devicons/devicon/icons/react/react-original.svg"
        />
      </div>
      <div>
        <img
          alt="html5"
          src="https://cdn.jsdelivr.net/gh/devicons/devicon/icons/html5/html5-original.svg"
        />
      </div>
      <div>
        <img
          alt="css3"
          src="https://cdn.jsdelivr.net/gh/devicons/devicon/icons/css3/css3-original.svg"
        />
      </div>
      <div>
        <img
          alt="postgresql"
          src="https://cdn.jsdelivr.net/gh/devicons/devicon/icons/postgresql/postgresql-original.svg"
        />
      </div>
      <div>
        <img
          alt="docker"
          src="https://cdn.jsdelivr.net/gh/devicons/devicon/icons/docker/docker-original.svg"
        />
      </div>
      <div>
        <img
          alt="amazonwebservices"
          src="https://cdn.jsdelivr.net/gh/devicons/devicon/icons/amazonwebservices/amazonwebservices-original.svg"
        />
      </div>
      <div>
        <img
          alt="git"
          src="https://cdn.jsdelivr.net/gh/devicons/devicon/icons/git/git-original.svg"
        />
      </div>
      <div>
        <img
          alt="jira"
          src="https://cdn.jsdelivr.net/gh/devicons/devicon/icons/jira/jira-original.svg"
        />
      </div>
      <img
        alt="asana"
        src="https://cdn.worldvectorlogo.com/logos/asana-logo.svg"
      />
    </div>
  </div>
</DocumentFragment>
`;
